core/src/main/scala/com/stripe/dagon/Dag.scala

```diff
@@ -622,10 +626,14 @@ sealed abstract class Dag[N[_]] extends Serializable { self =>
         case (_, _, Nil, Nil, _) =>
           None
         case (depth, _, Nil, nextBatch, seen) =>
-          // nextBatch has at least one item, and sorting preserves that
-          val h :: tail = sort(nextBatch)
-          val (nextBatch1, seen1) = computeNext(nextBatch, seen)
-          Some((depth + 1, h, tail, nextBatch1, seen1))
+          sort(nextBatch) match {
+            case h :: tail =>
+              val (nextBatch1, seen1) = computeNext(nextBatch, seen)
+              Some((depth + 1, h, tail, nextBatch1, seen1))
+            case Nil =>
+              // nextBatch has at least one item, and sorting preserves that
+              sys.error("impossible")
```

Comment:
We could remove the previous case and just return None here. That was we don’t need to explain why it is safe and we actually do fewer matches.

Reply:
Previous case being line 628 in the new code.

Reply:
👍